operators/src/adapters/raster_time.rs

```diff
@@ -0,0 +1,379 @@
+    // TODO: calculate at start when tiling info is available before querying first tile
+    num_spatial_tiles: Option<usize>,
+    current_spatial_tile: usize,
+    #[pin]
+    stream: Zip<St1, St2>,
+    ended: bool,
+}
+
+impl<T1, T2, St1, St2, F1, F2> RasterTimeAdapter<T1, T2, St1, St2, F1, F2>
+where
+    T1: Pixel,
+    T2: Pixel,
+    St1: Stream<Item = Result<RasterTile2D<T1>>>,
+    St2: Stream<Item = Result<RasterTile2D<T2>>>,
+    F1: Fn(QueryRectangle) -> St1,
+    F2: Fn(QueryRectangle) -> St2,
+{
+    pub fn new(source_a: F1, source_b: F2, query_rect: QueryRectangle) -> Self {
+        Self {
+            stream: source_a(query_rect).zip(source_b(query_rect)),
+            source_a,
+            source_b,
+            query_rect,
+            num_spatial_tiles: None,
+            current_spatial_tile: 0,
+            time_end: None,
+            ended: false,
+        }
+    }
+
+    // TODO: move to TimeInterval
+    fn intersect_time_intervals(t1: TimeInterval, t2: TimeInterval) -> Option<TimeInterval> {
+        if t1.intersects(&t2) {
+            let start = std::cmp::max(t1.start, t2.start);
+            let end = std::cmp::min(t1.end, t2.end);
+            Some(TimeInterval::new_unchecked(start, end))
+        } else {
+            None
+        }
+    }
+
+    fn align_tiles(
+        mut tile_a: RasterTile2D<T1>,
+        mut tile_b: RasterTile2D<T2>,
+    ) -> (RasterTile2D<T1>, RasterTile2D<T2>) {
+        // TODO: scale data if measurement unit requires it?
+        let time = Self::intersect_time_intervals(tile_a.time, tile_b.time)
+            .expect("intervals must overlap");
+        tile_a.time = time;
+        tile_b.time = time;
+        (tile_a, tile_b)
+    }
+
+    fn number_of_tiles_in_bbox(tile_info: &TileInformation, bbox: BoundingBox2D) -> usize {
+        // TODO: get tiling strategy from stream or execution context instead of creating it here
+        let strat = TilingStrategy {
+            bounding_box: bbox,
+            tile_pixel_size: tile_info.tile_size_in_pixels,
+            geo_transform: tile_info.global_geo_transform,
+        };
+
+        strat.tile_grid_box().number_of_elements()
+    }
+}
+
+impl<T1, T2, St1, St2, F1, F2> Stream for RasterTimeAdapter<T1, T2, St1, St2, F1, F2>
+where
+    T1: Pixel,
+    T2: Pixel,
+    St1: Stream<Item = Result<RasterTile2D<T1>>>,
+    St2: Stream<Item = Result<RasterTile2D<T2>>>,
+    F1: Fn(QueryRectangle) -> St1,
+    F2: Fn(QueryRectangle) -> St2,
+{
+    type Item = Result<(RasterTile2D<T1>, RasterTile2D<T2>)>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let RasterTimeAdapterProjection {
+            source_a,
+            source_b,
+            query_rect,
+            time_end,
+            num_spatial_tiles,
+            current_spatial_tile,
+            mut stream,
+            ended,
+        } = self.project();
+
+        if *ended {
+            return Poll::Ready(None);
+        }
```

Comment:
Maybe you can implement `FusedStream`, use `ended` there, and use the ` is_terminated(&self)` here.

Reply:
I tried but it wouldn't let me:
> error[E0382]: borrow of moved value: `self`
   --> operators\src\adapters\raster_time.rs:111:12
    |
99  |     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
    |                  ---- move occurs because `self` has type `Pin<&mut raster_time::RasterTimeAdapter<T1, T2, St1, St2, F1, F2>>`, which does not implement the `Copy` trait
...
109 |         } = self.project();
    |                  --------- `self` moved due to this method call
110 | 
111 |         if self.is_terminated() {
    |            ^^^^ value borrowed here after move
    |
note: this function consumes the receiver `self` by taking ownership of it, which moves `self`
   --> operators\src\adapters\raster_time.rs:18:1

Reply:
lets look together into it

Reply:
9d6080c75e065be0264b666d22a1d8d08e4bda2f